rsciio/digitalsurf/_api.py

```diff
@@ -909,35 +956,150 @@ def _build_original_metadata(
+            "magnification": SEM.get("Real Magnification"),
+            "microscope": "Attolight Allalin",
+            "Stage": {
+                "rotation": STAGE_IMAGE.get("stage_rotation_z"),
+                "rotation_units": "deg",
+                "tilt_alpha": STAGE_IMAGE.get("stage_rotation_x"),
+                "tilt_alpha_units": "deg",
+                "tilt_beta": STAGE_IMAGE.get("stage_rotation_y"),
+                "tilt_beta_units": "deg",
+                "x": STAGE_IMAGE.get("stage_position_x"),
+                "x_units": "mm",
+                "y": STAGE_IMAGE.get("stage_position_y"),
+                "y_units": "mm",
+                "z": STAGE_IMAGE.get("stage_position_z"),
+                "z_units": "mm",
+            }
+        }
+
+        return SEM_metadata
+
+    def _map_Spectrometer_metadata(self) -> dict:
+        """return Spectrometer metadata according to lumispy specifications"""
+        atto_omd = self.signal_dict['original_metadata']
+        # get nested dictionaries in an error-handling way
+        atto_omd = atto_omd.get('Object_0_Channel_0', {})
+        atto_omd = atto_omd.get('Parsed', {})
+        if atto_omd is None:
+            return {}
+        else:
+            spectrometer = atto_omd.get('SPECTROMETER', {})
+
+        Spectrometer_metadata = {
+            # "model":
+            # "acquisition_mode": ,
+            "entrance_slit_width": spectrometer.get('Entrance slit width'),
+            "entrance_slit_width_units": spectrometer.get('Entrance slit width_units'),
+            "exit_slit_width": spectrometer.get('Exit slit width'),
+            "exit_slit_width_units": spectrometer.get('Exit slit width_units'),
+            "central_wavelength": spectrometer.get('Central wavelength'),
+            "central_wavelength_units": spectrometer.get('Central wavelength_units'),
+            # "start_wavelength(nm)":
+            # "step_size(nm)"
+            "Grating": spectrometer.get('Grating'),
+            "groove_density": spectrometer.get('Grating - Groove Density'),
+            "groove_density_units": spectrometer.get('Grating - Groove Density_units'),
+            "blazing_wavelength": spectrometer.get('Grating - Blaze Angle'),
+            "blazing_wavelength_units": spectrometer.get('Central wavelength_units'),
+            "Filter": {
+                "filter_type": spectrometer.get('Filter')
+            }
+        }
+
+        return Spectrometer_metadata
+
+    def _map_spectral_detector_metadata(self) -> dict:
+        """return Spectrometer metadata according to lumispy specifications"""
+
+        atto_omd = self.signal_dict['original_metadata']
+        # get nested dictionaries in an error-handling way
+        atto_omd = atto_omd.get('Object_0_Channel_0', {})
+        atto_omd = atto_omd.get('Parsed', {})
+        if atto_omd is None:
+            return {}
+        else:
+            CCD = atto_omd.get('CCD', {})
+
+        spectral_detector_metadata = {
```

Comment:
The Attolight Metadata files contain additional Information about the CCD like Amplification and ReadoutRate - maybe one should add some of them here?

Reply:
Currently (even before this PR), all of this is saved `sig.original_metadata`. In this PR, part of it is mapped to `sig.metadata` according to the lumispy specification which does not include the fields Amplification or ReadoutRate: https://docs.lumispy.org/en/latest/user_guide/metadata_structure.html

Personally, I am of the opinion that metadata standardisation should be only performed if it brings added value to the project, ie in view of implementing analysis or visualisation functionality that require information about the dataset to be stored in a determined location. Do you have any idea for a function that would make use of these added fields? If they are here to be informative to the user I think they can stay in `original_metadata`.

Don't get me wrong, in the end if it makes you happy I will duplicate everything you want in `sig.metadata` :]

Reply:
Well, the LumiSpy specification was done with some other data files as example, so it could be amended where it makes sense. I think `amplification` could be useful, while `readout-rate` does not seem necessary indeed.

Apart from that, it is true that the mechanism for metadata mapping is now included in the reader and could be amended anytime if someone wishes to include other fields.

Reply:
I propose that we keep it true to the lumispy metadata structure and I will commit to keep it updated.